Datahub.Web/Pages/Search.cshtml

```diff
@@ -39,18 +39,18 @@
 
     <div>
         <ul class="pagination">
-            <li class="page-item @(Model.CurrentPage == 1 ? "disabled":"")">
-                <a data-page="1" class="page-link" href="/search?q=@(Model.QueryString)&p=1"><i class="fas fa-fast-backward"></i></a>
+            <li class="page-item @(Model.SearchParams.p == 1 ? "disabled":"")">
+                <a data-page="1" class="page-link" href="/search?q=@(Model.SearchParams.q)&p=1"><i class="fas fa-fast-backward"></i></a>
             </li>
-            <li class="page-item @(Model.CurrentPage == 1 ? "disabled":"")">
-                <a class="page-link" href="/search?q=@(Model.QueryString)&p=@(Model.CurrentPage - 1)"><i class="fas fa-step-backward"></i></a>
+            <li class="page-item @(Model.SearchParams.p == 1 ? "disabled":"")">
+                <a class="page-link" href="/search?q=@(Model.SearchParams.q)&p=@(Model.SearchParams.p - 1)"><i class="fas fa-step-backward"></i></a>
             </li>
-            <li class="page-item  @(Model.CurrentPage == (Math.Ceiling((decimal) Model.Results.Total / Model.CurrentPageSize)) ? "disabled":"")">
-                <a class="page-link" href="/search?q=@(Model.QueryString)&p=@(Model.CurrentPage + 1)"><i class="fas fa-step-forward"></i></a>
+            <li class="page-item  @(Model.SearchParams.p == (Math.Ceiling((decimal) Model.Results.Total / Model.SearchParams.size)) ? "disabled":"")">
+                <a class="page-link" href="/search?q=@(Model.SearchParams.q)&p=@(Model.SearchParams.p + 1)"><i class="fas fa-step-forward"></i></a>
             </li>
 
-            <li class="page-item @(Math.Ceiling((decimal) Model.Results.Total / Model.CurrentPageSize) == Model.CurrentPage ? "disabled": "")">
-                <a href="/search?q=@(Model.QueryString)&p=@(Math.Ceiling((decimal) Model.Results.Total / Model.CurrentPageSize))" class="page-link"><i class="fas fa-fast-forward"></i></a>
+            <li class="page-item @(Math.Ceiling((decimal) Model.Results.Total / Model.SearchParams.size) == Model.SearchParams.p ? "disabled": "")">
+                <a href="/search?q=@(Model.SearchParams.q)&p=@(Math.Ceiling((decimal) Model.Results.Total / Model.SearchParams.size))" class="page-link"><i class="fas fa-fast-forward"></i></a>
             </li>
```

Comment:
Still not sold on this paging code, I know I wrote it, but feels like we could do better, I couldn't find a library to wrap it up easily, not high priority obviously but would like to revisit this at some point

Reply:
Reminds me that there is a paging bug that Felix found last week. Don't think we've repro'd yet. Obvs this was just your PoC code but feel free to make a ticket and rework if you think like 😄